galasa-managers-parent/galasa-managers-core-parent/dev.galasa.textscan.manager/src/main/java/dev/galasa/textscan/ITextScanner.java

```diff
@@ -0,0 +1,422 @@
+     * @return This text scanner for fluent calls
+     * @throws FailTextFoundException - If the failurePattern was found
+     * @throws MissingTextException - If no occurrences of the searchPattern was found 
+     */
+    ITextScanner scan(String text, Pattern searchPattern, Pattern failPattern) throws FailTextFoundException, MissingTextException;
+    
+    /**
+     * Convenience method for scan(text, searchPattern, null, 1)
+     * 
+     * @param text - The text to be searched
+     * @param searchPattern - The regex to search for
+     * @return This text scanner for fluent calls
+     * @throws MissingTextException - If no occurrences of the searchPattern was found 
+     */
+    ITextScanner scan(String text, Pattern searchPattern) throws MissingTextException;
+    
+    /**
+     * Convenience method for scan(text, Pattern.Compile("\Q" + searchString + "\E"), Pattern.Compile("\Q" + failString + "\E"), count)
+     * 
+     * @param text - The text to be searched
+     * @param searchText - The text to search for
+     * @param failText - Failure text to search for, can be null meaning no fail search
+     * @param count - Atleast how many occurrences of the searchText must exist
+     * @return This text scanner for fluent calls
+     * @throws FailTextFoundException - If the failureText was found
+     * @throws MissingTextException - If no occurrences of the searchText was found 
+     * @throws IncorrectOccurancesException - If insufficient occurrences were found, if there are zero occurrences, then MissingTextException will be thrown
+     */
+    ITextScanner scan(String text, String searchString, String failString, int count) throws FailTextFoundException, MissingTextException, IncorrectOccurancesException;
+    
+    /**
+     * Convenience method for scan(text, searchString, failString, 1)
+     * 
+     * @param text - The text to be searched
+     * @param searchText - The text to search for
+     * @param failText - Failure text to search for
+     * @return This text scanner for fluent calls
+     * @throws FailTextFoundException - If the failureText was found
+     * @throws MissingTextException - If no occurrences of the searchText was found 
+     */
+    ITextScanner scan(String text, String searchString, String failString) throws FailTextFoundException, MissingTextException;
+    
+    /**
+     * Convenience method for scan(text, searchString, null, 1)
+     * 
+     * @param text - The text to be searched
+     * @param searchText - The text to search for
+     * @return This text scanner for fluent calls
+     * @throws MissingTextException - If no occurrences of the searchText was found 
+     */
+    ITextScanner scan(String text, String searchString) throws MissingTextException;
+    
+    
+    
+    
+    
+    
+    
+    
+    /**
+     * Search a IScannable for regex patterns.  It will search initially search for any occurrence of the failPattern before searching for the searchPattern.
```

Comment:
why is one of the arguments the thing you want to scan?  Is this consistent with the rest of the methods?

Reply:
Not sure you have read this correctly.

Reply:
This comment is due to confusion on where the code is implemented.   as Managers should stick to their own areas,  managers like batch should not be providing any text scanning code,  just code to update and refresh the logs.